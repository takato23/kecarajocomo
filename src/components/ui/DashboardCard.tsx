import { ReactNode } from 'react';

import { cn } from '@/lib/utils';

interface DashboardCardProps {
  title?: string;
  subtitle?: string;
  children: ReactNode;
  className?: string;
  action?: ReactNode;
  loading?: boolean;
  error?: string;
}

export function DashboardCard({
  title,
  subtitle,
  children,
  className,
  action,
  loading = false,
  error
}: DashboardCardProps) {
  return (
    <div className={cn(
      "bg-white rounded-lg shadow-sm border border-gray-200 p-6",
      "hover:shadow-md transition-shadow duration-200",
      className
    )}>
      {(title || action) && (
        <div className="flex items-center justify-between mb-4">
          <div>
            {title && (
              <h3 className="text-lg font-semibold text-gray-900">{title}</h3>
            )}
            {subtitle && (
              <p className="text-sm text-gray-600 mt-1">{subtitle}</p>
            )}
          </div>
          {action && (
            <div className="flex-shrink-0">
              {action}
            </div>
          )}
        </div>
      )}
      
      {loading ? (
        <div className="flex items-center justify-center py-8">
          <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-primary-600"></div>
        </div>
      ) : error ? (
        <div className="flex items-center justify-center py-8">
          <div className="text-center">
            <p className="text-red-600 text-sm font-medium">Error loading data</p>
            <p className="text-gray-500 text-xs mt-1">{error}</p>
          </div>
        </div>
      ) : (
        children
      )}
    </div>
  );
}