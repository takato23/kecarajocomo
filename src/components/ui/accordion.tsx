import * as React from "react"
import { ChevronDown } from "lucide-react"

import { cn } from "@/lib/utils"

const Accordion = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, ...props }, ref) => (
  <div ref={ref} className={cn("space-y-2", className)} {...props} />
))
Accordion.displayName = "Accordion"

interface AccordionItemProps extends React.HTMLAttributes<HTMLDivElement> {
  value: string
}

const AccordionItem = React.forwardRef<HTMLDivElement, AccordionItemProps>(
  ({ className, ...props }, ref) => (
    <div
      ref={ref}
      className={cn("border rounded-lg", className)}
      {...props}
    />
  )
)
AccordionItem.displayName = "AccordionItem"

const AccordionTrigger = React.forwardRef<
  HTMLButtonElement,
  React.ButtonHTMLAttributes<HTMLButtonElement> & {
    className?: string
    children?: React.ReactNode
  }
>(({ className, children, ...props }, ref) => (
  <button
    ref={ref}
    className={cn(
      "flex w-full items-center justify-between p-4 font-medium transition-all hover:bg-muted/50 [&[data-state=open]>svg]:rotate-180",
      className
    )}
    {...props}
  >
    {children}
    <ChevronDown className="h-4 w-4 shrink-0 transition-transform duration-200" />
  </button>
))
AccordionTrigger.displayName = "AccordionTrigger"

const AccordionContent = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, children, ...props }, ref) => (
  <div
    ref={ref}
    className={cn(
      "overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down",
      className
    )}
    {...props}
  >
    <div className="pb-4 pt-0 px-4">{children}</div>
  </div>
))
AccordionContent.displayName = "AccordionContent"

export { Accordion, AccordionItem, AccordionTrigger, AccordionContent }