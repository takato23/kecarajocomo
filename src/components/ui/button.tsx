import * as React from "react"
import { Slot } from "@radix-ui/react-slot"
import { cva, type VariantProps } from "class-variance-authority"

import { cn } from "@/lib/utils"

const buttonVariants = cva(
  "inline-flex items-center justify-center whitespace-nowrap rounded-2xl text-sm font-semibold ring-offset-background transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 shadow-sm hover:shadow-md active:scale-95",
  {
    variants: {
      variant: {
        default: "bg-gradient-to-r from-emerald-500 to-teal-600 text-white hover:from-emerald-600 hover:to-teal-700",
        destructive:
          "bg-gradient-to-r from-red-500 to-pink-600 text-white hover:from-red-600 hover:to-pink-700",
        outline:
          "border-2 border-gray-200 bg-white hover:bg-gray-50 hover:border-gray-300 dark:border-gray-700 dark:bg-gray-800 dark:hover:bg-gray-700",
        secondary:
          "bg-gradient-to-r from-gray-100 to-gray-200 text-gray-900 hover:from-gray-200 hover:to-gray-300 dark:from-gray-700 dark:to-gray-600 dark:text-white",
        ghost: "hover:bg-gray-100 hover:text-gray-900 dark:hover:bg-gray-800 dark:hover:text-white",
        link: "text-emerald-600 underline-offset-4 hover:underline dark:text-emerald-400",
        premium: "bg-gradient-to-r from-purple-500 to-pink-600 text-white hover:from-purple-600 hover:to-pink-700",
      },
      size: {
        default: "h-12 px-6 py-3",
        sm: "h-10 rounded-xl px-4 text-xs",
        lg: "h-14 rounded-2xl px-8 text-base",
        xl: "h-16 rounded-3xl px-10 text-lg",
        icon: "h-12 w-12",
      },
    },
    defaultVariants: {
      variant: "default",
      size: "default",
    },
  }
)

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    VariantProps<typeof buttonVariants> {
  asChild?: boolean
}

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  ({ className, variant, size, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : "button"
    return (
      <Comp
        className={cn(buttonVariants({ variant, size, className }))}
        ref={ref}
        {...props}
      />
    )
  }
)
Button.displayName = "Button"

export { Button, buttonVariants } 