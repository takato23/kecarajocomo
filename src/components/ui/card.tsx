import * as React from "react"

import { cn } from "@/lib/utils"

const Card = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn(
      "rounded-3xl border border-gray-200/50 bg-white/80 backdrop-blur-sm text-card-foreground shadow-lg hover:shadow-xl transition-all duration-300 dark:border-gray-700/50 dark:bg-gray-800/80",
      className
    )}
    {...props}
  />
))
Card.displayName = "Card"

const CardHeader = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn("flex flex-col space-y-2 p-8", className)}
    {...props}
  />
))
CardHeader.displayName = "CardHeader"

const CardTitle = React.forwardRef<
  HTMLParagraphElement,
  React.HTMLAttributes<HTMLHeadingElement>
>(({ className, ...props }, ref) => (
  <h3
    ref={ref}
    className={cn(
      "text-2xl font-bold leading-none tracking-tight bg-gradient-to-r from-gray-900 to-gray-600 bg-clip-text text-transparent dark:from-white dark:to-gray-300",
      className
    )}
    {...props}
  />
))
CardTitle.displayName = "CardTitle"

const CardDescription = React.forwardRef<
  HTMLParagraphElement,
  React.HTMLAttributes<HTMLParagraphElement>
>(({ className, ...props }, ref) => (
  <p
    ref={ref}
    className={cn("text-sm text-gray-600 dark:text-gray-400", className)}
    {...props}
  />
))
CardDescription.displayName = "CardDescription"

const CardContent = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, ...props }, ref) => (
  <div ref={ref} className={cn("p-8 pt-0", className)} {...props} />
))
CardContent.displayName = "CardContent"

const CardFooter = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn("flex items-center p-8 pt-0", className)}
    {...props}
  />
))
CardFooter.displayName = "CardFooter"

export { Card, CardHeader, CardFooter, CardTitle, CardDescription, CardContent } 