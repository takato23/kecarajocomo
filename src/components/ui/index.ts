export { DashboardCard } from './DashboardCard';
export { MetricDisplay } from './MetricDisplay';
export { QuickActionCard } from './QuickActionCard';

// shadcn/ui components
export * from './alert';
export * from './badge';
export { Button, buttonVariants } from './button';
export * from './card';
export * from './dialog';
export * from './dropdown-menu';
export * from './input';
export * from './label';
export * from './scroll-area';
export * from './select';
export * from './separator';
export * from './skeleton';
export * from './slider';
export * from './switch';
export * from './tabs';
export * from './textarea';