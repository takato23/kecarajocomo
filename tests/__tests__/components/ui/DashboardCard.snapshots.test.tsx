import React from 'react';
import { render } from '@testing-library/react';
import { DashboardCard } from '@/components/ui/DashboardCard';

// Mock button component for actions
const MockButton = ({ children }: { children: React.ReactNode }) => (
  <button className="mock-button">{children}</button>
);

// Mock content component
const MockContent = ({ title }: { title: string }) => (
  <div className="mock-content">
    <h4>{title}</h4>
    <p>This is mock content for testing.</p>
  </div>
);

describe('DashboardCard Snapshots', () => {
  describe('Basic DashboardCard', () => {
    it('should render basic dashboard card with content only', () => {
      const { container } = render(
        <DashboardCard>
          <MockContent title="Basic Content" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render dashboard card with title', () => {
      const { container } = render(
        <DashboardCard title="Dashboard Title">
          <MockContent title="Content with Title" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render dashboard card with title and subtitle', () => {
      const { container } = render(
        <DashboardCard 
          title="Dashboard Title" 
          subtitle="This is a subtitle"
        >
          <MockContent title="Content with Title and Subtitle" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('DashboardCard with Actions', () => {
    it('should render dashboard card with action button', () => {
      const { container } = render(
        <DashboardCard 
          title="Dashboard Title"
          action={<MockButton>Action</MockButton>}
        >
          <MockContent title="Content with Action" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render dashboard card with action but no title', () => {
      const { container } = render(
        <DashboardCard action={<MockButton>Action</MockButton>}>
          <MockContent title="Content with Action Only" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render dashboard card with multiple actions', () => {
      const { container } = render(
        <DashboardCard 
          title="Dashboard Title"
          action={
            <div className="flex gap-2">
              <MockButton>Edit</MockButton>
              <MockButton>Delete</MockButton>
            </div>
          }
        >
          <MockContent title="Content with Multiple Actions" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('DashboardCard Loading States', () => {
    it('should render loading state', () => {
      const { container } = render(
        <DashboardCard 
          title="Loading Card"
          loading={true}
        >
          <MockContent title="This should not be visible" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render loading state with action', () => {
      const { container } = render(
        <DashboardCard 
          title="Loading Card"
          subtitle="Loading subtitle"
          loading={true}
          action={<MockButton>Action</MockButton>}
        >
          <MockContent title="This should not be visible" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render loading state without title', () => {
      const { container } = render(
        <DashboardCard loading={true}>
          <MockContent title="This should not be visible" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('DashboardCard Error States', () => {
    it('should render error state', () => {
      const { container } = render(
        <DashboardCard 
          title="Error Card"
          error="Failed to load data"
        >
          <MockContent title="This should not be visible" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render error state with action', () => {
      const { container } = render(
        <DashboardCard 
          title="Error Card"
          subtitle="Error subtitle"
          error="Network connection failed"
          action={<MockButton>Retry</MockButton>}
        >
          <MockContent title="This should not be visible" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render error state without title', () => {
      const { container } = render(
        <DashboardCard error="Something went wrong">
          <MockContent title="This should not be visible" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render error state with detailed message', () => {
      const { container } = render(
        <DashboardCard 
          title="Detailed Error"
          error="Unable to connect to the server. Please check your internet connection and try again."
        >
          <MockContent title="This should not be visible" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('DashboardCard Custom Styling', () => {
    it('should render dashboard card with custom className', () => {
      const { container } = render(
        <DashboardCard 
          title="Custom Card"
          className="custom-dashboard-card"
        >
          <MockContent title="Custom styled content" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render dashboard card with multiple custom classes', () => {
      const { container } = render(
        <DashboardCard 
          title="Multi-class Card"
          className="custom-class-1 custom-class-2 border-red-500"
        >
          <MockContent title="Multi-class content" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('DashboardCard Complex Content', () => {
    it('should render dashboard card with complex content structure', () => {
      const { container } = render(
        <DashboardCard 
          title="Complex Card"
          subtitle="With complex content"
          action={<MockButton>View All</MockButton>}
        >
          <div className="space-y-4">
            <div className="grid grid-cols-2 gap-4">
              <div className="bg-gray-50 p-3 rounded">
                <h5 className="font-medium">Metric 1</h5>
                <p className="text-2xl font-bold">123</p>
              </div>
              <div className="bg-gray-50 p-3 rounded">
                <h5 className="font-medium">Metric 2</h5>
                <p className="text-2xl font-bold">456</p>
              </div>
            </div>
            <div className="text-sm text-gray-600">
              Last updated: 2 minutes ago
            </div>
          </div>
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render dashboard card with list content', () => {
      const { container } = render(
        <DashboardCard 
          title="List Card"
          subtitle="Recent activities"
        >
          <ul className="space-y-2">
            <li className="flex items-center justify-between p-2 bg-gray-50 rounded">
              <span>Activity 1</span>
              <span className="text-sm text-gray-500">2 min ago</span>
            </li>
            <li className="flex items-center justify-between p-2 bg-gray-50 rounded">
              <span>Activity 2</span>
              <span className="text-sm text-gray-500">5 min ago</span>
            </li>
            <li className="flex items-center justify-between p-2 bg-gray-50 rounded">
              <span>Activity 3</span>
              <span className="text-sm text-gray-500">10 min ago</span>
            </li>
          </ul>
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render dashboard card with chart-like content', () => {
      const { container } = render(
        <DashboardCard 
          title="Chart Card"
          subtitle="Performance metrics"
          action={<MockButton>Details</MockButton>}
        >
          <div className="space-y-4">
            <div className="flex items-center justify-between">
              <span className="text-sm text-gray-600">Revenue</span>
              <span className="font-semibold">$12,345</span>
            </div>
            <div className="w-full bg-gray-200 rounded-full h-2">
              <div className="bg-blue-600 h-2 rounded-full" style={{ width: '70%' }}></div>
            </div>
            <div className="flex items-center justify-between">
              <span className="text-sm text-gray-600">Growth</span>
              <span className="font-semibold text-green-600">+12%</span>
            </div>
          </div>
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('DashboardCard Edge Cases', () => {
    it('should render dashboard card with empty content', () => {
      const { container } = render(
        <DashboardCard title="Empty Card">
          <div></div>
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render dashboard card with very long title', () => {
      const { container } = render(
        <DashboardCard 
          title="This is a very long title that might wrap to multiple lines or be truncated"
          subtitle="This is also a long subtitle that provides additional context"
        >
          <MockContent title="Long title content" />
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render dashboard card with special characters', () => {
      const { container } = render(
        <DashboardCard 
          title="Special Characters: éñüñ & 中文 🎉"
          subtitle="Testing unicode & emojis"
        >
          <div>Content with special chars: àáâãäåæçèéêë</div>
        </DashboardCard>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});