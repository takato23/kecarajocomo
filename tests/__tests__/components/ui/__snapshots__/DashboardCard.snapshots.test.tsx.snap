// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardCard Snapshots Basic DashboardCard should render basic dashboard card with content only 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="mock-content"
  >
    <h4>
      Basic Content
    </h4>
    <p>
      This is mock content for testing.
    </p>
  </div>
</div>
`;

exports[`DashboardCard Snapshots Basic DashboardCard should render dashboard card with title 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Dashboard Title
      </h3>
    </div>
  </div>
  <div
    class="mock-content"
  >
    <h4>
      Content with Title
    </h4>
    <p>
      This is mock content for testing.
    </p>
  </div>
</div>
`;

exports[`DashboardCard Snapshots Basic DashboardCard should render dashboard card with title and subtitle 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Dashboard Title
      </h3>
      <p
        class="text-sm text-gray-600 mt-1"
      >
        This is a subtitle
      </p>
    </div>
  </div>
  <div
    class="mock-content"
  >
    <h4>
      Content with Title and Subtitle
    </h4>
    <p>
      This is mock content for testing.
    </p>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Complex Content should render dashboard card with chart-like content 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Chart Card
      </h3>
      <p
        class="text-sm text-gray-600 mt-1"
      >
        Performance metrics
      </p>
    </div>
    <div
      class="flex-shrink-0"
    >
      <button
        class="mock-button"
      >
        Details
      </button>
    </div>
  </div>
  <div
    class="space-y-4"
  >
    <div
      class="flex items-center justify-between"
    >
      <span
        class="text-sm text-gray-600"
      >
        Revenue
      </span>
      <span
        class="font-semibold"
      >
        $12,345
      </span>
    </div>
    <div
      class="w-full bg-gray-200 rounded-full h-2"
    >
      <div
        class="bg-blue-600 h-2 rounded-full"
        style="width: 70%;"
      />
    </div>
    <div
      class="flex items-center justify-between"
    >
      <span
        class="text-sm text-gray-600"
      >
        Growth
      </span>
      <span
        class="font-semibold text-green-600"
      >
        +12%
      </span>
    </div>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Complex Content should render dashboard card with complex content structure 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Complex Card
      </h3>
      <p
        class="text-sm text-gray-600 mt-1"
      >
        With complex content
      </p>
    </div>
    <div
      class="flex-shrink-0"
    >
      <button
        class="mock-button"
      >
        View All
      </button>
    </div>
  </div>
  <div
    class="space-y-4"
  >
    <div
      class="grid grid-cols-2 gap-4"
    >
      <div
        class="bg-gray-50 p-3 rounded"
      >
        <h5
          class="font-medium"
        >
          Metric 1
        </h5>
        <p
          class="text-2xl font-bold"
        >
          123
        </p>
      </div>
      <div
        class="bg-gray-50 p-3 rounded"
      >
        <h5
          class="font-medium"
        >
          Metric 2
        </h5>
        <p
          class="text-2xl font-bold"
        >
          456
        </p>
      </div>
    </div>
    <div
      class="text-sm text-gray-600"
    >
      Last updated: 2 minutes ago
    </div>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Complex Content should render dashboard card with list content 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        List Card
      </h3>
      <p
        class="text-sm text-gray-600 mt-1"
      >
        Recent activities
      </p>
    </div>
  </div>
  <ul
    class="space-y-2"
  >
    <li
      class="flex items-center justify-between p-2 bg-gray-50 rounded"
    >
      <span>
        Activity 1
      </span>
      <span
        class="text-sm text-gray-500"
      >
        2 min ago
      </span>
    </li>
    <li
      class="flex items-center justify-between p-2 bg-gray-50 rounded"
    >
      <span>
        Activity 2
      </span>
      <span
        class="text-sm text-gray-500"
      >
        5 min ago
      </span>
    </li>
    <li
      class="flex items-center justify-between p-2 bg-gray-50 rounded"
    >
      <span>
        Activity 3
      </span>
      <span
        class="text-sm text-gray-500"
      >
        10 min ago
      </span>
    </li>
  </ul>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Custom Styling should render dashboard card with custom className 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200 custom-dashboard-card"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Custom Card
      </h3>
    </div>
  </div>
  <div
    class="mock-content"
  >
    <h4>
      Custom styled content
    </h4>
    <p>
      This is mock content for testing.
    </p>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Custom Styling should render dashboard card with multiple custom classes 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border p-6 hover:shadow-md transition-shadow duration-200 custom-class-1 custom-class-2 border-red-500"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Multi-class Card
      </h3>
    </div>
  </div>
  <div
    class="mock-content"
  >
    <h4>
      Multi-class content
    </h4>
    <p>
      This is mock content for testing.
    </p>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Edge Cases should render dashboard card with empty content 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Empty Card
      </h3>
    </div>
  </div>
  <div />
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Edge Cases should render dashboard card with special characters 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Special Characters: Ã©Ã±Ã¼Ã± & ä¸­æ–‡ ðŸŽ‰
      </h3>
      <p
        class="text-sm text-gray-600 mt-1"
      >
        Testing unicode & emojis
      </p>
    </div>
  </div>
  <div>
    Content with special chars: Ã Ã¡Ã¢Ã£Ã¤Ã¥Ã¦Ã§Ã¨Ã©ÃªÃ«
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Edge Cases should render dashboard card with very long title 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        This is a very long title that might wrap to multiple lines or be truncated
      </h3>
      <p
        class="text-sm text-gray-600 mt-1"
      >
        This is also a long subtitle that provides additional context
      </p>
    </div>
  </div>
  <div
    class="mock-content"
  >
    <h4>
      Long title content
    </h4>
    <p>
      This is mock content for testing.
    </p>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Error States should render error state 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Error Card
      </h3>
    </div>
  </div>
  <div
    class="flex items-center justify-center py-8"
  >
    <div
      class="text-center"
    >
      <p
        class="text-red-600 text-sm font-medium"
      >
        Error loading data
      </p>
      <p
        class="text-gray-500 text-xs mt-1"
      >
        Failed to load data
      </p>
    </div>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Error States should render error state with action 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Error Card
      </h3>
      <p
        class="text-sm text-gray-600 mt-1"
      >
        Error subtitle
      </p>
    </div>
    <div
      class="flex-shrink-0"
    >
      <button
        class="mock-button"
      >
        Retry
      </button>
    </div>
  </div>
  <div
    class="flex items-center justify-center py-8"
  >
    <div
      class="text-center"
    >
      <p
        class="text-red-600 text-sm font-medium"
      >
        Error loading data
      </p>
      <p
        class="text-gray-500 text-xs mt-1"
      >
        Network connection failed
      </p>
    </div>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Error States should render error state with detailed message 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Detailed Error
      </h3>
    </div>
  </div>
  <div
    class="flex items-center justify-center py-8"
  >
    <div
      class="text-center"
    >
      <p
        class="text-red-600 text-sm font-medium"
      >
        Error loading data
      </p>
      <p
        class="text-gray-500 text-xs mt-1"
      >
        Unable to connect to the server. Please check your internet connection and try again.
      </p>
    </div>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Error States should render error state without title 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-center py-8"
  >
    <div
      class="text-center"
    >
      <p
        class="text-red-600 text-sm font-medium"
      >
        Error loading data
      </p>
      <p
        class="text-gray-500 text-xs mt-1"
      >
        Something went wrong
      </p>
    </div>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Loading States should render loading state 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Loading Card
      </h3>
    </div>
  </div>
  <div
    class="flex items-center justify-center py-8"
  >
    <div
      class="animate-spin rounded-full h-8 w-8 border-b-2 border-primary-600"
    />
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Loading States should render loading state with action 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Loading Card
      </h3>
      <p
        class="text-sm text-gray-600 mt-1"
      >
        Loading subtitle
      </p>
    </div>
    <div
      class="flex-shrink-0"
    >
      <button
        class="mock-button"
      >
        Action
      </button>
    </div>
  </div>
  <div
    class="flex items-center justify-center py-8"
  >
    <div
      class="animate-spin rounded-full h-8 w-8 border-b-2 border-primary-600"
    />
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard Loading States should render loading state without title 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-center py-8"
  >
    <div
      class="animate-spin rounded-full h-8 w-8 border-b-2 border-primary-600"
    />
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard with Actions should render dashboard card with action but no title 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div />
    <div
      class="flex-shrink-0"
    >
      <button
        class="mock-button"
      >
        Action
      </button>
    </div>
  </div>
  <div
    class="mock-content"
  >
    <h4>
      Content with Action Only
    </h4>
    <p>
      This is mock content for testing.
    </p>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard with Actions should render dashboard card with action button 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Dashboard Title
      </h3>
    </div>
    <div
      class="flex-shrink-0"
    >
      <button
        class="mock-button"
      >
        Action
      </button>
    </div>
  </div>
  <div
    class="mock-content"
  >
    <h4>
      Content with Action
    </h4>
    <p>
      This is mock content for testing.
    </p>
  </div>
</div>
`;

exports[`DashboardCard Snapshots DashboardCard with Actions should render dashboard card with multiple actions 1`] = `
<div
  class="bg-white rounded-lg shadow-sm border border-gray-200 p-6 hover:shadow-md transition-shadow duration-200"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div>
      <h3
        class="text-lg font-semibold text-gray-900"
      >
        Dashboard Title
      </h3>
    </div>
    <div
      class="flex-shrink-0"
    >
      <div
        class="flex gap-2"
      >
        <button
          class="mock-button"
        >
          Edit
        </button>
        <button
          class="mock-button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
  <div
    class="mock-content"
  >
    <h4>
      Content with Multiple Actions
    </h4>
    <p>
      This is mock content for testing.
    </p>
  </div>
</div>
`;
